### PR TITLE
`@types/greasemonkey`: add `responseType` property to `GM.Request` 

### PR DESCRIPTION
**Pull request by trim21**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes

<https://wiki.greasespot.net/GM.xmlHttpRequest>

### CODE REVIEW

File: types/greasemonkey/index.d.ts [hunk 1]
```diff
@@ -105,8 +105,8 @@ declare namespace GM {
         user?: string | undefined;
         /** Password to use for authentication purposes */
         password?: string | undefined;
-        /** Decode the response as specified type **/
-        responseType: "" | "arraybuffer" | "blob" | "document" | "json" | "text" | "stream" | undefined;
+        /** Decode the response as specified type */
```
Comment (nikolay-borzov) [inline types/greasemonkey/index.d.ts:108]: Could you indicate that default value is "text"?
```
Decode the response as specified type. Default value is "text"
```

**Review by MysteryBlokHed — APPROVED** [review]
LGTM!

File: types/greasemonkey/index.d.ts [hunk 2]
```diff
@@ -105,6 +105,8 @@ declare namespace GM {
         user?: string | undefined;
         /** Password to use for authentication purposes */
         password?: string | undefined;
+        /** Decode the response as specified type. Default value is "text" */
+        responseType?: XMLHttpRequestResponseType | undefined;
```
Comment (nikolay-borzov) [inline types/greasemonkey/index.d.ts:109]: Maybe we don't need `| undefined` if value is already optional `?:`

**Review by nikolay-borzov — APPROVED** [review]

File: types/greasemonkey/index.d.ts (hunk 2 above)
Comment (Trim21) [reply types/greasemonkey/index.d.ts:109]: I'm not sure about this, other properties all have `| undefined` so I add it.

**Comment by MysteryBlokHed** [comment]
The `| undefined` is still useful for compatibility with TypeScript 4.4’s
exact optional properties option

**Comment by Trim21** [comment]
when can this PR get merged？

**Comment by nikolay-borzov** [comment]
Whenever you want. See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60608#issuecomment-1144002248

Ready to merge

**Comment by nikolay-borzov** [comment]
Ready to merge

**Comment by Trim21** [comment]
sorry，didn't notice that I can metge it by myself through bot.

**Comment by nikolay-borzov** [comment]
Thanks for the contribution!

**Pull request merged**